numpy/core/overrides.py

```diff
@@ -213,6 +223,9 @@ def decorator(implementation):
         if module is not None:
             public_api.__module__ = module
 
+        if docs_from_dispatcher and dispatcher.__doc__:
```

Comment:
The `and dispatcher.__doc__` seems necessary; strangely enough there's unprotected uses of `dispatcher.__doc__` a little higher up in `array_function_dispatch` that don't seem to cause any issues. Unrelated to this PR, but we may want to check that.